src/components/editor/ArticleListController.jsx

```diff
@@ -0,0 +1,109 @@
+import FalcorController from "lib/falcor/FalcorController";
+import List from "./List";
+
+var pageLength = 50;
+
+export default class ArticleListController extends FalcorController {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleUpdateSlugValue = this.handleUpdateSlugValue.bind(this);
+    this.safeSetState({
+      slugSearchValue: ""
+    })
+  }
+  static getFalcorPath(params) {
+    return [['articlesByTime', 'paginated', {from: pageLength*(params.page-1), length: pageLength}, ['title', 'slug']], ['articlesByTime', 'paginated', 'total']];
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let pathStart = "/articles/page/" + this.props.params.page + "/article/slug/";
+    browserHistory.push(pathStart + this.state.slugSearchValue + "/");
+    this.safeSetState({
+      slugSearchValue: ""
+    });
+  }
+
+  handleUpdateSlugValue(event) {
+    this.safeSetState({
+      slugSearchValue: event.target.value
+    });
+  }
+
+  handlePageChange(increment, event) {
+    let path = this.props.location.pathname;
+    path = path.split('/');
+    path[3] = (parseInt(path[3])+increment).toString();
+    browserHistory.push(path.join('/'));
+  }
+
+  render() {
+      // For Ling when code reviewing. I was choosing between two style decisions here when making the loading look as I wanted it
+      // I could either do as I did now with quite a few ternary operators. Or I could've put it all in a big if this.state.ready
+      // So that it only processed the data if it had it. But in that case I then just had to put in quite a few var's instead of let's
+      // So they would carry over to the return statement. Which would you think is best?
+      let data = this.state.data ? this.state.data.articlesByTime.paginated : null;
+      let page = this.props.params.page;
+      let maxPage = data ? Math.floor(data.total/pageLength) + (data.total%pageLength > 0) : null;
+      let firstIndex = pageLength*(page-1);
+
+      // If trying to access inacessible page, redirect to page 1
+      if (page < 1 || page > maxPage) {
+        return <p>You have tried accessing a page that doesn't exist. Please press <a href="/articles/page/1/">this link</a> to return to page 1.
+          If you believe this was unintended and there is an error with the website please contact the web development team of The Gazelle.</p>;
+      }
+
+      let indexConverter = data ? [...Array(Math.min(pageLength, data.total-firstIndex-5)).keys()].map((val) => {
+        return val+firstIndex
+      }) : null;
+      let pathStart = "/articles/page/" + page + "/";
+
+    return (
+      <div style={{height: "100%", padding: 0, margin: 0}}>
+        <div style={{height: "100%", width: "50%", float: "left"}}>
+          <h3>Articles</h3>
+          <p style={{width: "75%"}}>Here the 50 most recent articles are shown in the list, if you have an older article in mind,
```

Comment:
Yeah I completely agree it would be a great user experience. And yeah if we implement it all in one go, I'd definitely love to include it as well.

So I'll do exactly that, leave it as it is right now, and upgrade it later, together with the add author one.